vim-patch:8.1.2313,8.2.{2046,2769}

janlazo opened the pull request.

janlazo committed:
vim-patch:8.1.2313: debugging where a delay comes from is not easy

Problem:    Debugging where a delay comes from is not easy.
Solution:   Use different values when calling ui_delay().
https://github.com/vim/vim/commit/eda1da0c9a8db1400649629117e7d248c07735f7

janlazo committed:
vim-patch:8.2.2046: some test failures don't give a clear error

Problem:    Some test failures don't give a clear error.
Solution:   Use assert_match() and assert_fails() instead of assert_true().
            (Ken Takata, closes vim/vim#7368)
https://github.com/vim/vim/commit/5dc4e2f883896c99ebe83355822ac6067970b031

janlazo commented on src/nvim/testdir/test_options.vim:
```diff
@@ -621,6 +621,26 @@ func Test_local_scrolloff()
+  set writedelay=0
+  " With 'writedelay' set should take at least 30 * 10 msec
+  " call assert_inrange(30 * 0.01, 999.0, elapsed)
+  call assert_inrange(0.01, 999.0, elapsed)
```
@bfredl Is this suppose to happen?

bfredl replied:
'writedelay' works quite differently in neovim so not necessarily.

bfredl commented on src/nvim/buffer.c:
```diff
@@ -1844,7 +1844,7 @@ buf_T *buflist_new(char_u *ffname_arg, char_u *sfname_arg, linenr_T lnum,
       if (emsg_silent == 0) {
         ui_flush();
-        os_delay(3000L, true);  // make sure it is noticed
+        os_delay(3001L, true);  // make sure it is noticed
```
What is the rationale between these magic numbers and does it even apply to us?

janlazo replied:
:man_shrugging: It's a vim patch. I omitted the change where `ui_delay` is logged. I don't know the alternative to Vim's `ch_log()`.

On Vim, `ui_delay()` runs either `gui_wait_for_chars()` or `mch_delay` (which runs `WaitForChar()`) so the extra milliseconds matter.

janlazo replied:
Updated the patch to use `DLOG` instead of  `ch_log`.

janlazo committed:
vim-patch:8.2.2769: Modula-3 config files are not recognized

Problem:    Modula-3 config files are not recognized.
Solution:   Add filetype patterns. (Doug Kearns)
https://github.com/vim/vim/commit/6bc00699c5049c40761d212bef1b490e2af7944c
